src/status_im/models/browser.cljs

```diff
@@ -31,8 +32,14 @@
       (assoc browser :dapp? true :name (:name dapp))
       (assoc browser :dapp? false :name (i18n/label :t/browser)))))
 
+(defn check-if-phishing-url [{:keys [history history-index] :as browser}]
+  (let [history-host (http/url-host (try (nth history history-index) (catch js/Error _)))]
+    (assoc browser :unsafe? (dependencies/phishing-detect history-host))))
```

Comment:
is it a pure function or a side-effect ?

Reply:
Why merge if it's not pure, you put that in an event, events are suppose to be pure...